src/routes/posts/+page.server.ts

```diff
@@ -3,18 +3,24 @@ import type { Article } from '$lib/types';
 
 export const load: PageServerLoad = async ({ platform, url }) => {
 	const page = parseInt(url.searchParams.get('page') ?? '0');
+	const category = url.searchParams.get('category')
+		? parseInt(url.searchParams.get('category') as string)
+		: null;
+	const start = url.searchParams.get('start') ?? '2022-12-31';
+	const end = url.searchParams.get('end') ?? new Date().toISOString().split('T')[0];
 
 	const response = await platform?.env.DB.prepare(
 		`SELECT aid FROM articles
-		WHERE images_published_with_alt_text != images_published
+		WHERE images_published_with_alt_text != images_published 
+		${category ? `AND categories LIKE "%${category}%"` : ''}
+		AND date >= "${start}" AND date <= "${end}"
```

Comment:
I think this line is vulnerable to SQL injection since the `url.searchParams.get` is not validated. I didn't make a note about the `page * 18` string interpolation because it is somewhat validated using `parseInt`. The preferred way to do this is with binding: <https://developers.cloudflare.com/d1/build-with-d1/d1-client-api/#parameter-binding>

Reply:
cringe moment of forgetting bindings exist